Lib/unittest/test/testmock/testpatch.py

```diff
@@ -664,6 +664,21 @@ def test():
         test()
 
 
+    def test_patch_dict_decorator_resolution(self):
+        # bpo-35512: Ensure that patch with a string target resolves to
+        # the new dictionary during function call
+        @patch.dict('unittest.test.testmock.support.target', {'bar': 'BAR'})
+        def test():
+            self.assertEqual(support.target['foo'], 'BAZ')
+            self.assertEqual(support.target['bar'], 'BAR')
+
+        support.target = {'foo': 'BAZ'}
+        test()
```

Comment:
I'd expect this `test()` call to be wrapped in a try/finally to put `support.target` back like it was, regardless of what happens in test.

Reply:
Yes, tests might use the variable in the future. Good catch. Thanks.